open-api/rest-catalog-open-api.yaml

```diff
@@ -846,6 +846,125 @@ paths:
         5XX:
           $ref: '#/components/responses/ServerErrorResponse'
 
+  /v1/{prefix}/transactions/commit:
+    parameters:
+      - $ref: '#/components/parameters/prefix'
+
+    post:
+      tags:
+        - Catalog API
+      summary: Commit updates to multiple tables in an all-or-nothing operation
+      operationId: commitTransaction
+      requestBody:
+        description:
+          Commit updates to multiple tables in an all-or-nothing operation
+
+
+          A commit for a single table consists of a table identifier with requirements and updates.
+          Requirements are assertions that will be validated before attempting to make and commit changes.
+          For example, `assert-ref-snapshot-id` will check that a named ref's snapshot ID has a certain value.
+
+
+          Updates are changes to make to table metadata. For example, after asserting that the current main ref
+          is at the expected snapshot, a commit may add a new child snapshot and set the ref to the new
+          snapshot id.
+
+
+          A server implementation needs to guarantee that updates across multiple tables are applied in an
+          all-or-nothing operation.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CommitTransactionRequest'
+        required: true
+      responses:
+        204:
+          description: Success, no content
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        403:
+          $ref: '#/components/responses/ForbiddenResponse'
+        404:
```

Comment:
Does this make sense with multiple tables? The route exists. Maybe this is a bad request?

Reply:
The only case where I thought it might make sense to throw a 404 is when the server gets a diff for a table that doesn't exist (anymore). Alternatively we could handle this case via a general `CommitFailedException`, but `CommitFailedException` is retryable

Reply:
I think it makes more sense to have a different error. The route does exist. Plus, our existing handlers don't know how to handle a 404 where the missing object is provided in the error message.

Reply:
I'm undecided. On one hand, that would already work and we wouldn't need to find a response code for "concurrent delete" that avoids the retry. On the other, the route doesn't represent that table.

I think I'm inclined to go with what you have here right now. It makes sense and we can always deprecate its use later.